lib/fluent/plugin/out_forward.rb

```diff
@@ -304,7 +304,7 @@ def send_data(node, tag, chunk)
         sock.write tag.to_msgpack  # tag
 
         # beginRaw(size)
-        sz = chunk.size
+        sz = chunk.size(decompressed: :text)
```

Comment:
The code in your patch generates tempfile twice. It's too heavy for almost workloads.